TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls

```diff
@@ -1,165 +1,50 @@
-                    if(tempMap.containskey(hcc.RecordTypeName__c)){
-                        System.debug(hcc.RecordTypeName__c);
-                        totalCost = tempMap.get(hcc.RecordTypeName__c);
-                        totalCost = Integer.valueof(totalCost + hcc.Cost__c);
-                        tempMap.put(hcc.RecordTypeName__c,totalCost);
-                    }else{
-                        tempMap.put(hcc.RecordTypeName__c,hcc.Cost__c);
-                    } 
-                    caseIdRecordTypeCostMap.put(hcc.Case2__c,tempMap);
-                }else{
-                    Map<String,Decimal> tempMap = new Map<String,Decimal>();
-                    tempMap.put(hcc.RecordTypeName__c,hcc.Cost__c);
-                    caseIdRecordTypeCostMap.put(hcc.Case2__c,tempMap);
-                }
-            }  
-        }
-        System.debug(caseIdRecordTypeSumMap);
-        Map<Id,Case> caseUpdateMap = new Map<Id,Case>();
-        for(Id key : caseIdRecordTypeSumMap.keySet()){
-            Case c = new Case();
-            c.id = key;
-            Map<String,Decimal> tempMap = caseIdRecordTypeSumMap.get(key);
-            for(String recordType : tempMap.keySet()){
-                if(recordType == 'Ambulance'){
-                    c.Ambulance_Total2__c = tempMap.get(recordType);
-                }else if(recordType == 'Pharmacare'){
-                    c.Pharmacare_Total2__c = tempMap.get(recordType);
-                }else if(recordType == 'Hospitalization'){
-                    c.Hospitalization_Total2__c = tempMap.get(recordType);                   
-                }
-                else if(recordType == 'MSP'){
-                    c.MSP_Total2__c = tempMap.get(recordType);
-                }
-            }    
-            caseUpdateMap.put(c.id,c);
-        }
-        for(Id key : caseIdRecordTypeCostMap.keySet()){
+        Map<Id,case> caseMap = new Map<Id,case>();
+        for(AggregateResult agg : groupedResults){
             Case c = new Case();
-            c.id = key;
-            Map<String,Decimal> tempMap = caseIdRecordTypeCostMap.get(key);
-            for(String recordType : tempMap.keySet()){
-                if(recordType == 'Continuing_Care'){
-                    c.Continuing_Care_Total2__c = tempMap.get(recordType);
-                }
-                if(recordType == 'Future_Care'){
-                    c.Future_Care_Total2__c = tempMap.get(recordType);
-                }
-            }    
-            Case c1 = caseUpdateMap.get(c.Id);
-            c1.Continuing_Care_Total2__c = c.Continuing_Care_Total2__c;
-            c1.Future_Care_Total2__c = c.Future_Care_Total2__c;
-            caseUpdateMap.put(c1.id,c1);
-        }
-        try{
-            update caseUpdateMap.values();
-        }catch(Exception e){
-            System.debug(e.getMessage());
-        }
-        
-    } 
-    public static void updateRollupDelete(List<Healthcare_Cost__c> hccList){
```

Comment:
This code is removed as handling of delete separately is not requiring when using the aggregate query approach.